doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+and a detection predicate. Configuration are applied when their associated
+"scenario" is detected.
+
+"Scenario" are detected through arbitrary criterion provided by the PF host
+(see below).
+
+## Definition
+
+<dl>
+<dt>Configuration</dt>
+<dd>
+Set of values for different parameters. A configuration **MUST NOT** contain 2
+values of the same parameters.
+
+For example, given a PF with 3 integer parameters A,B,C, a configuration can
+contain:
+
+ - 1 value: (A) or (B) or (C); or
+ - 2 values: (A,B) or (A,C) or (B,C); or
+ - 3 values: (A,B,C).
+</dd>
+
+<dt>Rogue Parameter</dt>
+<dd>
+A Parameter that is not contained by any configuration.
+<dd>
+</dl>
+
+## Configuration
+
+A PF **MUST** offer configurations as described in the Definition chapter.
+<nb>rule based parameter engine does not manipulate directly values, it
+applies configuration on the parameters.</nb>
+<why>This is what the reference implementation does.</why>
+
+Each configuration **MUST** be associated with a predicate that condition its
+eligibility. A configuration with a predicate that evaluates to `true` is called
+an "eligible configuration"
+<why>This is what the reference implementation does.</why>
+
+It **SHOULD** be possible to express a predicate to always evaluates to `true`.
+<why>in order to have parameters set to constant values or have a fallback
+configuration in a domain -- see below.</why>
+
+The predicate **SHOULD** be a "selection criterion rule". See next chapter for a
+definition.
+<why>The reference implementation uses a boolean expression based engine.</why>
+
+## Selection criterion
+
+A selection criterion **MUST** have one, and only one, state at a given time.
+
+A selection criterion **MUST** know at construction all possible states.
+<why>To be able to validate: -- rules on start -- state changes</why>
+
+The selection criterion possible states **MUST** be specifiable by directly a
+state set (`Input -> states == identity`)
+<nb>called **exclusive criterion**</nb>
+<nb>An empty set is not allowed as the criterion could not have a state.</nb>
+<why>any criterion can be crated with this one.</why>
+
+The selection criterion possible states **SHOULD** be specifiable by a combination
+of values
+<nb>combination in the mathematical sense `"ab" -> ["", "a", "b", "ab"]`</nb>
+<nb>called **inclusive criterion**</nb>
+<nb>An empty value set is allowed as its combination -- a set containing the
+empty set -- would not be empty. The empty set would be the only possible
+criteria state.</nb>
+<why>The reference implementation supports it.</why>
+
+The PF **MUST NOT** limit the number of criteria.
+
+The PF **MUST NOT** limit the number of states of any given criterion
+<ko>The reference implementation only supports 32 values for an inclusive
+criterion and 2^32 values for an exclusive criterion</ko>
+
+### Definitions
+<dl>
+<dt>Selection criterion rule</dt>
+<dd>
+Function (in the mathematical sense) that **MUST** given selection criteria
+return a Boolean.
```

Comment:
It is a definition, not a requirement. See the "philosophy" section; a predicate is a function of the form `f: x --> boolean`: https://en.wikipedia.org/wiki/Predicate_%28mathematical_logic%29